service/lib/dinstaller/dbus/users.rb

```diff
@@ -90,13 +91,18 @@ def initialize(backend, logger)
         end
 
         dbus_method :SetFirstUser,
-          FUSER_SIG + ", out result:u" do |full_name, user_name, password, auto_login, data|
+          FUSER_SIG + ", out result:a{sv}" do |full_name, user_name, password, auto_login, data|
           logger.info "Setting first user #{full_name}"
-          backend.assign_first_user(full_name, user_name, password, auto_login, data)
+          issues = backend.assign_first_user(full_name, user_name, password, auto_login, data)
 
-          dbus_properties_changed(USERS_INTERFACE, { "FirstUser" => first_user }, [])
-          update_validation
-          0
+          if issues.empty?
+            dbus_properties_changed(USERS_INTERFACE, { "FirstUser" => first_user }, [])
+            update_validation
+          else
+            logger.info "First user fatal issues detected: #{issues.map(&:message)}"
+          end
+
+          [{ "result" => issues.empty? ? 0 : 1, "issues" => issues.map(&:message) }]
```

Comment:
Why not use a pair with the result and the issues, instead of a hash?

Reply:
And why not using a boolean instead of a number? D-Bus support a BOOLEAN type.

Reply:
> Why not use a pair with the result and the issues, instead of a hash?

Do not have an strong opinion, it was suggested by @joseivanlopez and was fine for me.

Reply:
> And why not using a boolean instead of a number? D-Bus support a BOOLEAN type.

The same here, I followed what looked like the convention in this class returning integers instead of booleans, we were also discussing that in some situation it will be not just true or false but the returned values could represent different kind of issues but as we are not in that point yet it makes sense to be a boolean.